app/graphql/mutations/digital_object/update_projects.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module Mutations
+  module DigitalObject
+    class UpdateProjects < Mutations::BaseMutation
+      argument :id, ID, required: true
+      argument :primary_project, Inputs::StringKey, required: true
+      argument(
+        :other_projects,
+        [Inputs::StringKey],
+        "An array representing the new set of other projects for the given object, "\
+        "or an empty array to remove all projects but the primary.",
+        required: true
+      )
+
+      field :digital_object, Types::DigitalObjectInterface, null: true
+      field :user_errors, [Types::Errors::FieldedInput], null: false
+
+      def resolve(id:, primary_project:, other_projects:)
+        digital_object = ::DigitalObject.find_by_uid!(id)
+        primary_project_change = proposed_primary_project_change(digital_object, primary_project)
+        authorize_change!(digital_object.primary_project, primary_project_change)
+        other_project_change = proposed_other_projects_change(digital_object, primary_project, other_projects)
+
+        ActiveRecord::Base.transaction do
+          digital_object.primary_project = primary_project_change if primary_project_change
+          digital_object.other_projects = other_project_change if other_project_change
+          digital_object.updated_by = context[:current_user]
+          digital_object.save
+        end
+        projects_update_response(digital_object)
+      end
+
+      def proposed_primary_project_change(digital_object, proposed_string_key)
+        Project.find_by!(string_key: proposed_string_key) unless digital_object.primary_project.string_key.eql(proposed_string_key)
+      end
+
+      def authorize_change!(current_primary, proposed_primary)
+        if proposed_primary
+          ability.authorize! :delete_objects, current_primary
+          ability.authorize! :create_objects, proposed_primary
+        else
+          ability.authorize! :update_objects, current_primary
+        end
```

Comment:
If a user is updating both the primary project and other_projects during the same request, this won't verify that they have update permission.  Update permission is required for modifying other_projects.

Reply:
I am hypothesizing that no one has `delete` but not `update` to save some work, but they are technically separate permissions.

Reply:
Updated to check permissions for `update` and `delete` if both the primary and other projects are changing.